### ISSUE TITLE
Minimal package setup for local installation

### ISSUE THREAD

**Issue by henry-prior** [google-deepmind/emergent_in_context_learning#1]
In order to run locally there are assumptions that these modules can be installed as a package. To get this working I moved all of the directories in this repo into a single `emergent_in_context_learning` directory (as a sub-directory of the repo directory with the same name). I also added a minimal `setup.py` as an entry point for pip.

Here's a diff for reference: https://github.com/deepmind/emergent_in_context_learning/compare/main...henry-prior:minimal-package-setup
I assume if you want to add this to the repo you'll want to add more to `setup.py` including authors, email, license, etc. It might also make sense to fold the `requirements.txt` into `setup.py` as dependencies, but not that important.

**Comment by scychan** [comment]
Hi Henry! Could you elaborate a bit more on what you mean that "there are assumptions that these modules can be installed as a package"? Thanks!

**Comment by henry-prior** [comment]
Sure! I actually just double checked something that I didn't try originally that worked without having to modify anything in the repo so I'll explain where the confusion was.

When I first cloned the repo I navigated to the repo folder before installing dependencies and trying to run the first example config, i.e. I was in the `emergent_in_context_learning` folder, which resulted in `Error while finding module specification for 'emergent_in_context_learning.experiment.experiment' (ModuleNotFoundError: No module named 'emergent_in_context_learning')`

I realize the issue now was that the expectation is to run everything one directory up, i.e. from the directory you cloned the repo into, hence the `__init__.py` in the base directory in the repo. The alternative I was suggesting was to group everything in the repo into another `emergent_in_context_learning` folder which would then allow us to install it as a package in our environment and be able to run it from any location using `python -m emergent_in_context_learning.experiment.experiment`.

TL;DR: False alarm, only suggestion would be to note in the `README` that everything should be executed one directory above the repo!

**Comment by scychan** [comment]
Great! Yes, I suspected that might have been the case -- I'll add a brief
line to clarify. Thanks and good luck with your experiments!

**Comment by henry-prior** [comment]
Glad we got it sorted, going to close the issue. Thanks!
